dev-client/src/store/sync/PushDispatcher.tsx

```diff
@@ -62,18 +66,21 @@ export const PushDispatcher = () => {
     /* Dispatch a push if needed */
     if (needsPush) {
       dispatchPush()
-        /* If the initial push failed, begin a retry cycle */
         .then(result => {
           if (!result.payload || 'error' in result.payload) {
+            /* If the initial push failed, begin a retry cycle */
             beginRetry();
+          } else if (Object.keys(result.payload.errors).length > 0) {
+            /* If the push yielded sync errors, notify the user */
+            syncNotifications.showError();
```

Comment:
Confirming: the difference here is that `payload.error` indicates a generic error communicating with the backend, but `payload.errors` indicate an error like "does not exist" or "not allowed" (as set by `mutationResponseToResults`), right?

Reply:
If that's the case, if we receive sync errors in the response from a retry cycle, the error wouldn't show since this is not in the retry interval's `dispatchPush`, but it should, right?

Reply:
Discussed on a call -- now I understand the problem, will update the retry mechanism to include the error handler correctly.